src/plugins/orangehrmRecruitmentPlugin/Api/Model/CandidateDetailedModel.php

```diff
@@ -91,7 +91,8 @@ public function toArray(): array
                 ],
             'status' => is_null($candidateVacancy) ? null :
                 $candidateVacancy->getDecorator()->getCandidateVacancyStatus(),
-            'hasAttachment' => !is_null($candidateAttachment)
+            'hasAttachment' => !is_null($candidateAttachment),
```

Comment:
just noticed,
don't load attachment to check availability,
```php
$candidateAttachment = $this->getRecruitmentAttachmentService()
            ->getRecruitmentAttachmentDao()
->getCandidateAttachmentByCandidateId($this->candidate->getId());
```
since you are loading blob fields also, it may costly.
Develop another method inside Dao `hasCandidateAttachmentByCandidateId`. you can use execute count query and check $count > 0.

Also do we really need separate RecruitmentAttachmentService, can't we use CandidateService

Reply:
done. There are three services totally in recruitment plugin.

  - VacancyService
  - CandidateService
  - RecruitmentAttachmentService

All the attachment related things are handled in RecruitmentAttachmentService. This was the way it was done in 4X and I also used the same. I think it separates the concern of the services